### ISSUE TITLE
Describing yuv422p10le streams

### ISSUE THREAD

**Issue by Auliyaa** [sekrit-twc/zimg#172]
Hi,

I've been trying to use zimg to resize an input stream of YUV 422 planar 10-bits images but I'm failing to understand how to describe the input format.

I took a look at the api_example.cpp and I can't get a grasp of the subsample_w member inside the zimage_format struct.

I ended up with something like this for my in & out formats:

`zimgxx::zimage_format in_format;
  in_format.width = 3840;
  in_format.height = 2160;
  in_format.depth = 10;
  in_format.color_family = ZIMG_COLOR_YUV;
  in_format.pixel_type = ZIMG_PIXEL_WORD;
  in_format.matrix_coefficients = ZIMG_MATRIX_BT709;
  in_format.pixel_range = ZIMG_RANGE_LIMITED;
  in_format.subsample_w=1;
  in_format.subsample_h=1;

  zimgxx::zimage_format out_format;
  out_format.width = 960;
  out_format.height = 540;
  out_format.depth = 10;
  out_format.color_family = ZIMG_COLOR_YUV;
  out_format.pixel_type = ZIMG_PIXEL_WORD;
  out_format.matrix_coefficients = ZIMG_MATRIX_BT709;
  out_format.pixel_range = ZIMG_RANGE_LIMITED;
  out_format.subsample_w=1;
  out_format.subsample_h=1;`

I reused the allocate_buffer function from api_example but I'm not getting the expected plane size values (16588800, 8294400, 8294400). My guess is that my format is wrong. I still went on and manually allocated my buffers with the following code:

`std::pair<zimgxx::zimage_buffer, std::shared_ptr<void>> allocate_buffer(const zimgxx::zimage_format &format,
                                                                        size_t alignment)
{
  zimgxx::zimage_buffer buf;
  std::shared_ptr<void> ptr;
  ptr.reset(aligned_malloc(format.width*format.height*4, alignment), &aligned_free);

  buf.stride(0) = format.width*2;
  buf.stride(1) = format.width;
  buf.stride(2) = format.width;
  buf.mask(0) = ZIMG_BUFFER_MAX;
  buf.mask(1) = ZIMG_BUFFER_MAX;
  buf.mask(2) = ZIMG_BUFFER_MAX;
  buf.data(0) = static_cast<unsigned char *>(ptr.get());
  buf.data(1) = static_cast<unsigned char *>(ptr.get())+format.width*format.height*2;
  buf.data(2) = static_cast<unsigned char *>(ptr.get())+format.width*format.height*3;

  return {buf,ptr};
}`

Once my graph is processed, I see that the resulting image is missing the U & V component values for the bottom half of the image:

![image](https://user-images.githubusercontent.com/6253590/158402637-e82102d3-0701-4d40-b55f-937d89be4416.png)
![image](https://user-images.githubusercontent.com/6253590/158402679-42ef3597-7aca-45e9-a765-9ec2e5e1eced.png)

So basically my issue seems to be related to how the image_format is configured in order to described a YUV 4:2:2 planar 10-bits image. Could anyone please point me in the right direction ?

Thanks.

**Comment by Auliyaa** [comment]
Sorry for the disruption, I found my issue. In my testing, I forgot to revert subsampling_h to 0 instead of 1.

I am not a smart man :)
